src/PrestaShopBundle/Controller/Admin/Sell/Catalog/MonitoringController.php

```diff
@@ -0,0 +1,260 @@
+        $disabledProductGrid = $this->getDisabledProductGrid($disabledProductFilters);
+        $productWithoutImageGrid = $this->getProductWithoutImageGrid($productWithoutImageFilters);
+        $productWithoutDescriptionGrid = $this->getProductWithoutDescriptionGrid($productWithoutDescriptionFilters);
+        $productWithoutPriceGrid = $this->getProductWithoutPriceGrid($productWithoutPriceFilters);
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Monitoring/index.html.twig', [
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'empty_category_grid' => $this->presentGrid($emptyCategoryGrid),
+            'delete_category_form' => $deleteCategoryForm->createView(),
+            'product_with_combination_grid' => $this->presentGrid($productWithCombinationGrid),
+            'product_without_combination_grid' => $this->presentGrid($productWithoutCombinationGrid),
+            'disabled_product_grid' => $this->presentGrid($disabledProductGrid),
+            'product_without_image_grid' => $this->presentGrid($productWithoutImageGrid),
+            'product_without_description_grid' => $this->presentGrid($productWithoutDescriptionGrid),
+            'product_without_price_grid' => $this->presentGrid($productWithoutPriceGrid),
+        ]);
+    }
+
+    /**
+     * Provides filters functionality
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $gridIdentifiers = $this->identifySearchableGrid($request);
+
+        /** @var ResponseBuilder $responseBuilder */
+        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
+
+        return $responseBuilder->buildSearchResponse(
+            $gridIdentifiers['grid_definition'],
+            $request,
+            $gridIdentifiers['grid_id'],
+            'admin_monitoring_index'
+        );
+    }
+
+    /**
+     * Parses grid identifying parts from request in order to recognize which grid is being filtered
+     *
+     * @param Request $request
+     *
+     * @return array
+     */
+    private function identifySearchableGrid(Request $request)
+    {
+        $gridDefinition = 'prestashop.core.grid.definition.factory.monitoring.empty_category';
+        $gridId = EmptyCategoryGridDefinitionFactory::GRID_ID;
+
+        if ($request->request->has(ProductWithCombinationGridDefinitionFactory::GRID_ID)) {
+            $gridDefinition = 'prestashop.core.grid.definition.factory.monitoring.product_with_combination';
+            $gridId = ProductWithCombinationGridDefinitionFactory::GRID_ID;
+        }
+
+        if ($request->request->has(ProductWithoutCombinationGridDefinitionFactory::GRID_ID)) {
```

Comment:
shouldn't `elseif`'s be used in this case? You don't want to check further conditions when the first one already succeeded.

Reply:
You could define some kind of map here. E.g.:

```php
$map = [
  ProductWithoutCombinationGridDefinitionFactory::GRID_ID => [
    'grid_id' => ProductWithoutCombinationGridDefinitionFactory::GRID_ID,
    'grid_definition' => 'prestashop.core.grid.definition.factory.monitoring.product_without_combination',
  ],
  // ...    
];
```

This way it's easier to extend, then all you have to do is loop through your map until you find required item and return it right away.